tests/unittests/UT.common.mk

```diff
@@ -0,0 +1,100 @@
+#
+# Copyright (c) Microsoft. All rights reserved.
+#
+
+_UT_PROJECT_DIR := $(shell pwd)
+_UT_BASE_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
+_UT_OUT_DIR := $(_UT_BASE_DIR)/../../build/Tests/UnitTests/$(UT_NAME)/OSX
+_UT_OBJ_DIR := $(_UT_OUT_DIR)/Objects
+_UT_FILENAME := $(UT_NAME).UnitTests.exe
+
+all: $(_UT_OUT_DIR)/$(_UT_FILENAME)
+.PHONY: all
+
+CLANG := clang++
+OSX_VERSION_MIN := 10.10
+
+_UT_OBJECTS := $(addsuffix .o,$(addprefix $(_UT_OBJ_DIR)/,$(subst $(_UT_PROJECT_DIR)/,,$(UT_FILES))))
+
+# Common shims for all tests.
+_UT_OBJECTS += $(_UT_OBJ_DIR)/windows.o $(_UT_OBJ_DIR)/IwMalloc.o $(_UT_OBJ_DIR)/LoggingNative.o
+ifeq ($(strip $(filter EntryPoint.cpp,$(UT_FILES))),)
+_UT_OBJECTS += $(_UT_OBJ_DIR)/EntryPoint.o
+endif
+
+_INC := $(_UT_BASE_DIR)/../frameworks/include $(_UT_BASE_DIR)/../frameworks/gtest $(_UT_BASE_DIR)/../frameworks/gtest/include 
+_INC += $(_UT_BASE_DIR)/../frameworks/OSXShims/include $(_UT_BASE_DIR)/../../include/xplat
+
+CXXFLAGS := -std=c++14
+CFLAGS := $(foreach i,$(_INC),-I $(i))
+CFLAGS += -DTARGET_OS_MAC
+CFLAGS += -DWINOBJCRT_IMPEXP=
+CFLAGS += -DLOGGING_IMPEXP=
+CFLAGS += -mmacosx-version-min=$(OSX_VERSION_MIN)
+CFLAGS += $(UT_CFLAGS)
+
+_UT_FRAMEWORKS := CoreFoundation Foundation $(UT_FRAMEWORKS)
```

Comment:
Some of these can be empry; UT_FRAMEWORKS is one of them.

If `RESOURCES`, `RESOURCE_DIRS`, `FRAMEWORKS`, `CFLAGS`, or `LDFLAGS` are empty, they're ignored.